enzy_htp/mutation/api.py

```diff
@@ -367,8 +369,59 @@ def mutate_stru_with_tleap(
 
     return stru_cpy
 
+def mutate_stru_with_pymol(
+        stru: Structure,
+        mutant: List[Mutation],
+        in_place: bool = False,
+        ) -> Structure:
+    """mutate the {stru} to its {mutant} structure using PyMOL.
+    Args:
+        stru: the 'WT' structure
+        mutant: a list of Mutation() which describes a mutant to the 'WT'
+        in_place: if make the changes to the structure in-place 
+    Returns:
+        The mutated structure.
+    """
+
+    sp = PDBParser()
+    temp_file_name = fs.get_valid_temp_name("pymol_mutation_result.pdb")
+
+    # san check
+    for mut in mutant:
+        if not isinstance(mut, Mutation):
+            _LOGGER.error(
+                f"mutant takes only a list of Mutation(). Current mutant is: {mutant}")
+            raise TypeError
+
+    stru_cpy = copy.deepcopy(stru)
+
+    # 1. load stru into pymol
+    pi = interface.pymol
+    temp_session = pi.new_pymol_session()
+    pymol_obj_name, session = pi.load_enzy_htp_stru(stru_cpy, temp_session)
+
+    # 2. loop through mutants and apply each one
+    for mut in mutant:
+        pi.point_mutate(mut.get_position_key(), mut.get_target(), pymol_obj_name, session)
+
+    # 3. save to a structure. TODO(SS): add function in pymol interface for this?
```

Comment:
Yes, please make a general function in pymol interface for converting `objects in pymol sessions` to `enzy_htp.Structure`

Reply:
`retain_order` seems not necessary. It will make the saved PDB looks weird. You can try look into the saved PDB.

Reply:
It is pretty cool how enzy_htp can handle the weird looking PDB though. I looked into the get_structure again and indeed the algorithm is general enough to handle this.

Reply:
<img width="372" alt="image" src="https://github.com/ChemBioHTP/EnzyHTP/assets/36098720/537f3b27-7c57-4f27-b2cd-626f5e001997">
This is what retain_order enabled saving looks like

Reply:
When I remove retain_order, it looks much better:

![image](https://github.com/ChemBioHTP/EnzyHTP/assets/68860396/146e2b5a-d483-493e-aa04-0cd46e68b457)

However, the test function fails as the atoms are no longer in the same order as they were before. Is this okay? I can change the test if so.

Reply:
It really is! Good catch on the atom order. We will keep "retain_order" then.
Maybe add another test with a multichain structure where a mutation is deployed on chain B. In principle, it also works.

Reply:
Keeping "retain_order" on a multichain structure doesn't seem like it will work:

![image](https://github.com/ChemBioHTP/EnzyHTP/assets/68860396/86e88879-56c2-4c70-8f1c-b453abec5e43)

To solve this, I was going to just check the # of chains and turn "retain_order" off if it was >1, but I found another problem that I think has to do with how pymol2 saves its PDBs. Whenever it saves a structure with HETATMs and CONECTs, it gets rid of the CONECTs and changes the HETATMs to ATOMs. This causes the following error when enzy_htp tries to load it into a structure:

![image](https://github.com/ChemBioHTP/EnzyHTP/assets/68860396/c61ef3ca-363e-41f9-bd1b-3ec9e43a13e4)

When I save a structure with the GUI, this problem does not occur, as it keeps the HETATMs and CONECTs. Not sure if there's a cmd/param in cmd.save to fix this, but I am currently looking.

Reply:
Omg the behavior on multichain is ridiculous. My suggestion is get rid of retain_order for all cases and in your test, you can assert atoms in the same residue using set() to make them order less.
We will write another function supporting maximumly restoring the atom order together with the name fix function.

Reply:
Also if you dont use retain_order, will the error below still pop off?

Reply:
Yeah, it'll still give the same error, even if I don't mutate the WT at all

Reply:
But I think in a EnzyHTP saved PDB there should be nothing but ATOM records. Can you send me the PDB file from Pymol that causes this error?

Reply:
It's puo_put.pdb in the test/mutation/data folder

Reply:
Ohhhh I found the reason. We were not allowing multiple chain id to show up in the ATOM chain.
(The error information is wrong. My mistake.) Would you allow me to commit a change fixing this in your branch?

Reply:
I think a good plan is to add an option in get_structure about whether allowing it. It does not follow the defination of PDB but we can turn it on for rare cases like this

Reply:
Yeah that sounds good, you can commit the change

Reply:
Cool! I commit some changes that fixed the problem by enabling allow_multichain_in_atom=True (find reference in the test I added)
You can use this option for your export_to_pdb method